File: app/src/main/java/com/orlinskas/notebook/di/LiveDataModule.java
package com.orlinskas.notebook.di;

import androidx.lifecycle.MutableLiveData;

import com.orlinskas.notebook.Enums;
import com.orlinskas.notebook.mvvm.model.Day;
import com.orlinskas.notebook.mvvm.model.Notification;

import java.util.List;

import dagger.Module;
import dagger.Provides;

@NotificationScope
@Module
public class LiveDataModule {
    @Provides
    public MutableLiveData<Enum<Enums.DownloadStatus>> repositoryStatusData() {
        return new MutableLiveData<>();
    }

    @Provides
    public MutableLiveData<Enum<Enums.ConnectionStatus>> connectionStatusData() {
        return new MutableLiveData<>();
    }

    @Provides
    public MutableLiveData<List<Notification>> allNotificationsData() {
        return new MutableLiveData<>();
    }

    @Provides
    public MutableLiveData<List<Day>> daysData() {
        return new MutableLiveData<>();
    }
}

File: app/src/main/java/com/orlinskas/notebook/di/LiveDataModule.kt
package com.orlinskas.notebook.di

import androidx.lifecycle.MutableLiveData

import com.orlinskas.notebook.Enums
import com.orlinskas.notebook.mvvm.model.Day
import com.orlinskas.notebook.mvvm.model.Notification

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
class LiveDataModule {
    @Provides
    fun repositoryStatusData(): MutableLiveData<Enum<Enums.DownloadStatus>> {
        return MutableLiveData()
    }

    @Provides
    fun connectionStatusData(): MutableLiveData<Enum<Enums.ConnectionStatus>> {
        return MutableLiveData()
    }

    @Provides
    fun allNotificationsData(): MutableLiveData<List<Notification>> {
        return MutableLiveData()
    }

    @Provides
    fun daysData(): MutableLiveData<List<Day>> {
        return MutableLiveData()
    }
}
